### PR TITLE
Enhancement/703/multiple taps

### PR DESCRIPTION
**Pull request by hackel6**
#703

### CODE REVIEW

**Comment by EmmanuelP** [comment]
Hi,

Sorry for the late reply.

I don't think we can select gain or black level using defined integer values. There is no guarantee a selector value is always mapped to the same string.

arv_camera_select_gain and arv_camera_select_black_level should use a string as the selector parameter.

File: src/arvcamera.c [hunk 1]
```diff
@@ -1823,6 +1823,43 @@ arv_camera_get_gain_auto (ArvCamera *camera, GError **error)
...
+ *
+ * Configures Gain Selector feature.
+ *
+ * Since: 0.8.23
+ **/
+
+void
+arv_camera_select_gain (ArvCamera *camera, const char *gainSelector, GError **error) {
+	arv_camera_set_string (camera, "GainSelector", gainSelector, error);
+
```
Comment (EmmanuelP) [inline src/arvcamera.c:1840]: Opening bracket should on its own line. Closing bracket is missing.

File: src/arvcamera.c [hunk 2]
```diff
@@ -1898,6 +1935,44 @@ arv_camera_get_black_level (ArvCamera *camera, GError **error)
...
+ * @blackLevelSelector: black level selection
+ * @error: a #GError placeholder, %NULL to ignore
+ *
+ * Configures Black Level Selector feature.
+ *
+ * Since: 0.8.23
+ **/
+
+void
+arv_camera_select_black_level (ArvCamera *camera, const char *blackLevelSelector, GError **error) {
```
Comment (EmmanuelP) [inline src/arvcamera.c:1950]: Opening bracket should on its own line.

**Review by EmmanuelP — CHANGES_REQUESTED** [review]

File: src/arvcamera.c (hunk 1 above)
Comment (hackel6) [reply src/arvcamera.c]: thanks, working too fast.

File: src/arvcamera.c (hunk 2 above)
Comment (hackel6) [reply src/arvcamera.c]: thanks, working too fast.

**Comment by hackel6** [comment]
@EmmanuelP, is this at least more of what you had in mind? raw strings? i guess it doesn't buy you all that much now since it just wraps to the arv_camera_set_string() method.

**Review by EmmanuelP — CHANGES_REQUESTED** [review]
Sorry for the late review.

It looks good. Please update the Since statements in the function documentation.

**Comment by EmmanuelP** [comment]
Thanks.

**Pull request merged**